File: src/LeetCode/Algorithms/Java/Counting_Bits.java
package LeetCode.Algorithms.Java;

import java.util.Hashtable;

// https://leetcode.com/problems/counting-bits/
public class Counting_Bits {
	public static void main(String[] args) {
		int num = 5;
		int[] result = countBits(num);
		
		for (int n : result) {
			System.out.print(n + " ");
		}
		System.out.println();
	}
	
	public static int[] countBits(int num) {
		int[] result = new int[num + 1];
		
		for (int i = 0; i <= num; i++) {
			result[i] = Integer.bitCount(i);
		}
		
		return result;
	}
}

File: src/LeetCode/Algorithms/JavaScript/Counting_Bits.js
/**
 * https://leetcode.com/problems/counting-bits/
 * @param {number} num
 * @return {number[]}
 */
var countBits = function(num) {
	// const arr = new Uint8Array(num + 1);
	const arr = [0];
	
	for (let i = 0; i <= num; i++) {
		arr[i] = arr[i >> 1] + (i & 1);
	}
	
	return [...arr];
};

const num = 5;

console.log(countBits(num));